File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Loader/IDisplay.ts
import { FairyGUI } from "csharp";
import { Rect } from "../../../../Core/Define/Rect";
import Color from "../DataDefine/Color";

export interface IDisplay{
    X:number;
    Y:number;
    Rotation:number;
    ScaleX:number;
    ScaleY:number;
    SortingOrder:number;
    Visible:boolean;
    Color:Color;
    UnitComponent:any;
    Bounds : FairyGUI.GGraph;
    HealthComponent : any;
    Self:any;//可以保存自身Object.比如Spine你可以保存SkeletonAnimation.然后在别的地方你可以使用Self做一些特定操作
    FlipX:boolean;
    FlipY:boolean;
    Dispose();
    SetHitBox(rect:Rect);
    GetHitBox():Rect;
    SetParent(object:any);
    SetAnimation(animation:string, loop?:boolean) : any;
    AddAnimation(animation:string, loop?:boolean,delay?:number):any;
    PlayFastAttack();
    PlaySlowAttack();
    PlayHop();
    PlayJump();
    PlayFastShake(duration:number);
    PlaySlowShake(duration:number);
    PlayStagger();
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Loader/IDisplay.cs
using FairyGUI;
using ModTheSpire.UI.ViewComponent;
using UnityEngine;

namespace ModTheSpire.Unit.Loader
{
    public interface IDisplay
    {
        float X { get; set; }
        float Y { get; set; }
        Vector3 Rotation { get; set; }
        float ScaleX { get; set; }
        float ScaleY { get; set; }
        int SortingOrder { get; set; }
        bool Visible { get; set; }
        Color32 Color { get; set; }
        float Alpha { get; set; }
        GComponent UnitComponent { get; set; }
        FairyGUI.GGraph Bounds { get; set; }
        ViewComponent_HealthBar HealthComponent { get; set; }
        object Self { get; }
        GObject PowerComponent { get; set; }
        GObject NameComponent { get; set; }
        //Self;//可以保存自身Object.比如Spine你可以保存SkeletonAnimation.然后在别的地方你可以使用Self做一些特定操作
        bool FlipX { get; set; }
        bool FlipY { get; set; }
        void Dispose();
        void SetHitBox(Rect rect);
        Rect GetHitBox();
        void SetParent(GComponent p);
        object SetAnimation(string animation, bool loop = false);
        object AddAnimation(string animation, bool loop = false,float delay = 0);
        void PlayFastAttack();
        void PlaySlowAttack();
        void PlayHop();
        void PlayJump();
        void PlayFastShake(float duration);
        void PlaySlowShake(float duration);
        void PlayStagger();
    }
}